web/src/main/webapp/app/app.js

```diff
@@ -45,7 +45,14 @@
 
   // Cancel any logout requests if one exists
   window.onload = function() {
-    $.get('/api/anvandare/logout/cancel');
+    if (window.sessionStorage.getItem('launchId')) {
```

Comment:
See if it is possible to extract the logic to separate method(s) and reuse them in the three places that you add the header.